app/Support/DesktopRuntime.php

```
@@ -0,0 +1,76 @@
+        $host = $appUrl['host'] ?? 'localhost';
+        $port = $appUrl['port'] ?? ($isSecure ? 443 : 80);
+        $path = str_starts_with($path, '/') ? $path : "/{$path}";
+
+        if (app()->environment('local') || self::enabled()) {
+            return "{$wsProtocol}://{$host}:".self::websocketPort().$path;
+        }
+
+        $portSuffix = (($isSecure && $port == 443) || (! $isSecure && $port == 80)) ? '' : ":{$port}";
+
+        return "{$wsProtocol}://{$host}{$portSuffix}{$path}";
+    }
```

PHPMD (2.15.0)

[warning] The method websocketUrl() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)